sstables/sstables.hh

```diff
@@ -1032,9 +1032,17 @@ public:
 // This method validates both the full checksum and the per-chunk checksum
 // for the entire Data.db.
 //
-// Returns true if all checksums are valid.
+// Returns `valid` if all checksums are valid.
+// Returns `invalid` if at least one checksum is invalid.
+// Returns `no_checksum` if the sstable is uncompressed and does not have
+// a CRC component (CRC.db is missing from TOC.txt).
 // Validation errors are logged individually.
-future<bool> validate_checksums(shared_sstable sst, reader_permit permit);
+enum class validate_checksums_result {
+    invalid = 0,
+    valid = 1,
+    no_checksum = 2
```

Comment:
We should add a test for this. Can be a follow-up.

Reply:
We do have a test. It's the `test_validate_checksums`. With commit  https://github.com/scylladb/scylladb/pull/20207/commits/547ed16af3dd87869ffd023199a5b344becdf416 we cover the valid/invalid values, and with commit https://github.com/scylladb/scylladb/pull/20207/commits/21e0791cdd90a5e4403b7898ab4f5ddc89d0d269 we cover the no_checksum value.

Do you have something else in mind?

Reply:
Yes, that is what I wanted. Thanks.